In [9]:
# Loads the two files
# Joins the two files to one dataframe
# Calculates total sales by state using a groupby function
# Sales are currently in string values, requiring you to translate the value to a number so it can be calculated
# Identify the number of unique companies you do business with in each state
# To get an accurate count, you will want to use a built in option like pandas.Series.nunique()
# Complete any other analysis on the data that helps you illustrate the best state to setup a headquarters to support your customers and office
# Use markdown  cells and code cells to present your case and story with your own words and headlines to deliver your point

# The Final result should be:
# committed and pushed to your repository with all code cells run
# all formatting in business-acceptable context and full paragraphs using headers and normal text as appropriate in markdown cells
# Post the url to your new repository on github here once you have completed the assignment.

import pandas as pd

contracts = pd.read_csv("C:/Users/Michael Botelho/bip-python-unh/sales_analysis/data/contracts.csv", delimiter =',', encoding="utf-8-sig")
customers = pd.read_csv("C:/Users/Michael Botelho/bip-python-unh/sales_analysis/data/customers.csv", delimiter =',', encoding="utf-8-sig")


# Changing Data Types

customers['id'] = customers['id'].astype(str)
contracts['customer_id'] = contracts['customer_id'].astype(str)



# Merge/Join to one DataFrame

df = pd.merge(customers, contracts, left_on='id', right_on='customer_id')


# Fix Dollar Signs & Commas & Convert to Float

df['deposit'] = df['deposit'].replace('[$,()]', '', regex=True)
df['deposit'] = df['deposit'].astype(float)



# Calculate Total Sales By State

total_sales = df.groupby(['state'])['deposit'].agg('mean', 'sum')
pd.set_option('display.max_columns', None)
total_sales.columns = ['state', 'total sales']



############################################################################################################################
# Identify the number of unique companies you do business with in each state
import pandas as pd
unique_companies = df.drop_duplicates(subset=['company'])

unique_companies.company
pd.set_option('display.max_rows', 25)


unique_companies.company
len(unique_companies)
# or
pd.Series.nunique(unique_companies['company'])

############################################################################################################################

# To get an accurate count, you will want to use a built in option like pandas.Series.nunique()
# Complete any other analysis on the data that helps you illustrate the best state to setup a headquarters to support your customers and office
# Use markdown  cells and code cells to present your case and story with your own words and headlines to deliver your point

# df['company']
#pd.Series({c: df[c].unique() for c in df})


#df.groupby('company', 'state').count()
df

print(df.value_counts('state'))
unique_companies.value_counts('state')

unique_companies

df

# Below = Shows each row of the df created called unique_companies
unique_companies.company

state
TX    104
CA     88
FL     70
WA     54
NY     43
     ... 
MS      6
WV      5
SC      5
ME      4
AR      2
Length: 39, dtype: int64


0          Bubblebox
7               Izio
12      Twitterworks
18              Viva
21       Jabberstorm
           ...      
966            Meemm
969    Twitterbridge
976           Rhynyx
983            Vinte
990             Katz
Name: company, Length: 162, dtype: object

# How to see more rows (below):

In [6]:
pd.set_option('display.max_rows', 25)

# Consulting Data Analysis Week #2

### 1. Identify the number of unique companies you do business with in each state

In [4]:
import pandas as pd
unique_companies = df.drop_duplicates(subset=['company'])


len(unique_companies)
# or
pd.Series.nunique(unique_companies['company'])

162

### 2. Complete any other analysis on the data that helps you illustrate the best state to setup a headquarters to support your customers and office

Overall Dataframe with all values

In [5]:
# Overall Dataframe with all values
print(df.value_counts('state'))

state
TX    104
CA     88
FL     70
WA     54
NY     43
     ... 
MS      6
WV      5
SC      5
ME      4
AR      2
Length: 39, dtype: int64


Unique DataFrame with unique values for company

In [104]:
# Unique Dataframe, in case there were more deposits in companies within a certain state, which would throw off the numbers
unique_companies.value_counts('state')

state
TX    15
CA    14
FL    14
NY    10
WA     7
      ..
WV     1
KY     1
MA     1
ME     1
AK     1
Length: 37, dtype: int64

In [28]:
unique_companies
df




#total average sales by region per contracts by region


df
total_sales / df.value_counts('state')
df

total_sum = df.groupby(['state'])['deposit'].agg('sum')
total_sum

state
AK     489000.0
AL     508000.0
AR      94000.0
AZ    1032000.0
CA    4616000.0
        ...    
TX    6014000.0
VA    1965000.0
WA    2830000.0
WI    1258000.0
WV     285000.0
Name: deposit, Length: 39, dtype: float64

## Which State?
As we see through the results: Texas, California, Florida, and New York are the states where most companies are located. On the first original dataframe Washington has more values than New York, but on the unique dataframe we see that New York has 10 unique companies, while Washington has 7 unique companies. This likely occured because the companies in Washington have more rows or contracts and deposits compared to the companies in New York.

Overall, it seems that the best state to have a headquarters in would be Texas because they have the most companies within the state at 15 and they have the most total rows at 104, which means that Texas likely has the most total deposits and contracts as well.